cmd/minikube/cmd/start.go

```diff
@@ -260,6 +263,17 @@ func generateConfig(cmd *cobra.Command, kVersion string) (cfg.Config, error) {
 		}
 	}
 
+	// Feed Docker our host proxy environment by default, so that it can pull images
+	if _, ok := r.(*cruntime.Docker); ok {
+		if !cmd.Flags().Changed("docker-env") {
+			for _, k := range proxyVars {
+				if v := os.Getenv(k); v != "" {
+					dockerEnv = append(dockerEnv, fmt.Sprintf("%s=%s", k, v))
```

Comment:
so, potentially user can have multiple dockerEnv variables set.
In case, user can have NO_PROXY as well as "HTTPS_PROXY" set which shd error out right? or this handled later?

Reply:
> It would be nice to have something more portable as, well ?

Yes, I agree. This is an introductory PR that only solves the problem for most (99%?) of users who use the defaults. Other runtimes so far don't have a mechanism for introducing environment variables, but I suspect that #3835 should cover them nicely, which adds them to /etc/environment.

Reply:
If a user manually sets --docker-env, this PR is disabled. 

NO_PROXY and HTTPS_PROXY are complimentary, and thus plumbed into the environment as expected.

Reply:
cool.